src/core/theme/index.ts

```diff
@@ -1,73 +1,67 @@
-import {
-  spacing as spacingTokens,
-  SpacingProp,
-  SpacingDesignTokens,
-} from './spacing';
+import { colors as colorTokens, shadows, gradients } from './colors';
+import { suomifiDesignTokens } from 'suomifi-design-tokens';
+import { spacing as spacingTokens, SpacingProp } from './spacing';
+import { boxShadowFocus, absoluteFocus, noMouseFocus } from './focus';
 import { zindexes } from './zindexes';
 import { transitions } from './transitions';
 import { radius } from './radius';
-export { SpacingProp };
 
-export type SuomifiTokens = typeof importedTokens;
-export type DefaultSuomifiTokens = typeof defaultThemeTokens;
-export type ColorProp = keyof typeof importedTokens.colors;
-export type SuomifiTheme = ReturnType<typeof suomifiTheme>;
+export {
+  TypographyDesignTokens,
+  ColorDesignTokens,
+  SpacingDesignTokens,
+  RawDesignTokens,
+  RawColorDesignTokens,
+  RawSpacingDesignTokens,
+  RawTypographyDesignTokens,
+} from 'suomifi-design-tokens';
 
+export type ColorProp = keyof typeof desingTokens.colors;
 export type TypographyProp = keyof typeof suomifiDesignTokens.typography;
+export { SpacingProp };
 
-export { TypographyDesignTokens } from 'suomifi-design-tokens';
-
-export interface PartialTokens {
-  colors?: Partial<ColorDesignTokens>;
-  spacing?: Partial<SpacingDesignTokens>;
-  typography?: Partial<TypographyDesignTokens>;
-}
+export type SuomifiDesignTokens = typeof desingTokens;
+export type SuomifiTheme = typeof suomifiTheme;
 
-type DefaultInternalTokens = typeof internalTokens;
-const internalTokens = {
+const desingTokens = {
   shadows,
   gradients,
-  outlines,
   zindexes,
   transitions,
   radius,
-};
-
-const importedTokens = {
   colors: colorTokens,
   spacing: spacingTokens,
   typography: suomifiDesignTokens.typography,
   values: suomifiDesignTokens.values,
 };
 
-export const defaultThemeTokens = {
-  ...importedTokens,
-  ...internalTokens,
+const derivedTokens = {
+  focus: {
+    absoluteFocus: absoluteFocus(desingTokens),
+    boxShadowFocus: boxShadowFocus(desingTokens),
+    noMouseFocus,
+  },
+};
+
+export const suomifiTheme = {
+  // Get all design tokens
+  ...desingTokens,
+  // Get all derived tokens
+  ...derivedTokens,
 };
 
 /**
- *  Theme tokens and tokens as CSS
+ * Generate theme using provided tokens. Returns generated theme with provided tokens as CSS.
  * @param tokens SuomifiTokens with libraryTokenOverrides, defaults to suomifi-design-tokens
  * @param tokens.colors color tokens, defaults to suomifi-design-tokens colors
  * @param tokens.spacing spacing tokens, defaults to suomifi-design-tokens spacing
  * @param tokens.typography typography tokens, defaults to suomifi-design-tokens typography
  * @param tokens.values token values in object format, defaults to suomifi-design-tokens values, not connected to other params
  */
-export const suomifiTheme = (
-  props: Partial<DefaultInternalTokens & PartialTokens>,
-) => {
+export const getSuomifiTheme = (
```

Comment:
This is currently unused and probably will not work as expected. The tooling needs further improvements once a ThemeProvider will be introduced. The basic ides should be like this:
- Default tokens are static
- User can provide overriding tokens
- Derived tokens like focus are generated from default and provided tokens, but only when the tokens change on toplevel (e.g. Provider props change).
Most of the things above should already work.

Reply:
This has been implemented after the initial comment. Advanced merge is still missing, so it's not possible e.g. to provide custom colors for gradients and in addition one custom gradient.